ui/app/AppLayouts/Wallet/addaccount/stores/AddAccountStore.qml

```diff
@@ -0,0 +1,308 @@
+    property bool seedPhraseWord2Valid: false
+    property int seedPhraseWord2WordNumber: -1
+    property bool seedPhraseBackupConfirmed: false
+    property bool derivationPathOutOfTheDefaultStatusDerivationTreeConfirmed: false
+    property bool derivationPathOutOfTheDefaultStatusDerivationTree: root.addAccountModule?
+                                                                         !root.addAccountModule.derivationPath.startsWith(Constants.addAccountPopup.predefinedPaths.ethereum) ||
+                                                                         (root.addAccountModule.derivationPath.match(/'/g) || []).length !== 3 ||
+                                                                         (root.addAccountModule.derivationPath.match(/\//g) || []).length !== 5
+                                                                       : false
+
+    readonly property var derivationPathRegEx: /^(m\/44'\/)([0-9|'|\/](?!\/'))*$/
+    property string selectedRootPath: Constants.addAccountPopup.predefinedPaths.ethereum
+    readonly property var roots: [Constants.addAccountPopup.predefinedPaths.custom,
+        Constants.addAccountPopup.predefinedPaths.ethereum,
+        Constants.addAccountPopup.predefinedPaths.ethereumRopsten,
+        Constants.addAccountPopup.predefinedPaths.ethereumLedger,
+        Constants.addAccountPopup.predefinedPaths.ethereumLedgerLive
+    ]
+
+    function resetStoreValues() {
+        root.enteredSeedPhraseIsValid = false
+        root.enteredPrivateKeyIsValid = false
+        root.addingNewMasterKeyConfirmed = false
+        root.seedPhraseRevealed = false
+        root.seedPhraseWord1Valid = false
+        root.seedPhraseWord1WordNumber = -1
+        root.seedPhraseWord2Valid = false
+        root.seedPhraseWord2WordNumber = -1
+        root.seedPhraseBackupConfirmed = false
+        root.derivationPathOutOfTheDefaultStatusDerivationTreeConfirmed = false
+        root.selectedRootPath = Constants.addAccountPopup.predefinedPaths.ethereum
+
+        root.cleanPrivateKey()
+        root.cleanSeedPhrase()
+    }
+
+    function moduleInitialized() {
+        if (!root.addAccountModule) {
+            console.warn("addAccountModule not initialized")
+            return false
+        }
+        return true
+    }
+
+    function submitAddAccount(event) {
+        if (!root.moduleInitialized() || !root.primaryPopupButtonEnabled) {
+            return
+        }
+
+        if(!event) {
+            root.currentState.doPrimaryAction()
+        }
+        else if (event.key === Qt.Key_Return || event.key === Qt.Key_Enter) {
+            event.accepted = true
+            root.currentState.doPrimaryAction()
+        }
+    }
+
+    function getSeedPhrase() {
+        if (!root.moduleInitialized()) {
+            return
+        }
+        return root.addAccountModule.getSeedPhrase()
+    }
+
+    function changeSelectedOrigin(keyUid) {
+        if (!root.moduleInitialized()) {
```

Comment:
Not sure that I fully get this.

Reply:
I see this `mouleInitialized` check in many places in this file. Is it possible to have a requirement (and respect it) that the module is initialized before creating this instance so that all these checks won't be needed?